app/src/main/java/io/outblock/lilico/network/UserRegisterUtils.kt

```diff
@@ -105,21 +120,39 @@ private fun registerFirebase(user: RegisterResponse, callback: (isSuccess: Boole
     }
 }
 
-private suspend fun registerServer(username: String, wallet: HDWallet): RegisterResponse {
+private suspend fun registerServer(username: String, prefix: String): RegisterResponse {
     val deviceInfoRequest = DeviceInfoManager.getDeviceInfoRequest()
     val service = retrofit().create(ApiService::class.java)
+    val keyPair = KeyManager.generateKeyWithPrefix(prefix)
     val user = service.register(
         RegisterRequest(
             username = username,
-            accountKey = AccountKey(publicKey = wallet.getPublicKey(removePrefix = true)),
+            accountKey = AccountKey(publicKey = keyPair.public.toFormatString()),
             deviceInfo = deviceInfoRequest
-
         )
     )
     logd(TAG, user.toString())
     return user
 }
 
+fun generatePrefix(text: String): String {
+    val timestamp = System.currentTimeMillis().toString()
+    val combinedInput = "${text}_$timestamp"
+    val bytes = MessageDigest.getInstance("SHA-256").digest(combinedInput.toByteArray())
+    return bytes.joinToString("") { "%02x".format(it) }
+}
+
+fun formatPublicKey(publicKey: PublicKey?): String {
+    return (publicKey as? ECPublicKey)?.w?.let {
+        val bytes = asUnsignedByteArray(it.affineX) + asUnsignedByteArray(it.affineY)
+        BaseEncoding.base16().lowerCase().encode(bytes)
+    } ?: ""
+}
+
+fun formatSignData(text: String, domainTag: ByteArray = normalize("FLOW-V0.0-user")): ByteArray {
+    return domainTag + text.encodeToByteArray()
```

Comment:
Use Flow SDK method, no need write tag for itself.

There is `FlowSigner` Interface in SDK, using it. Instead